docs/the-new-architecture/cxx-custom-types.md

```diff
@@ -0,0 +1,284 @@
+   readonly passLargeNumber: (input: string) => number;
+   // ..
+```
+
+</TabItem>
+<TabItem value="flow">
+
+```js title="NativeSampleModule.js"
+   // ...
+   +passLargeNumber: (input: string) => number;
+   // ..
+```
+
+</TabItem>
+</Tabs>
+
+Declare it in your `NativeSampleModule.h` file and include the `Int64.h` header file:
+
+```cpp
+//...
+#include "Int64.h"
+//...
+int32_t passLargeNumber(jsi::Runtime& rt, int64_t input);
+```
+
+And implement it in `NativeSampleModule.cpp`:
+
+```cpp
+//...
+#include <cmath>
+//...
+int32_t NativeSampleModule::passLargeNumber(jsi::Runtime& rt, int64_t input) {
+    return std::cbrt(input);
+}
+```
+
+In your app you can call this new native function via:
+
+```js
+<Section title="Cxx TurboModule">
+  NativeSampleModule.passLargeNumber(...) ={' '}
+  {JSON.stringify(
+    NativeSampleModule.passLargeNumber('9223372036854775807')
+  )}
+</Section>
+```
+
+which should return `2097152`.
+
+## Any custom type
+
+Similar to the example above you can now write custom `bridging` functionality for any custom C++ type you want to expose to react-native. E.g., you can add support for `folly::StringPiece`, `QString`, `boost::filesystem::path`, `absl::optional` or any other type you need to support in your C++ Turbo Native Modules.
+
+```cpp title="Path.h"
+#pragma once
+
+#include <react/bridging/Bridging.h>
+#include <boost/filesystem.hpp>
+
+namespace facebook::react {
+
+template<>
+struct Bridging<boost::filesystem::path> {
+  static boost::filesystem::path fromJs(jsi::Runtime& rt, const std::string& value) { // auto-bridge from jsi::String to std::string
+    return boost::filesystem::path(value);
+  }
+
+  static jsi::String toJs(jsi::Runtime& rt, boost::filesystem::path value) {
+    return bridging::toJs(rt, value.string());
+  }
+};
+
+} // namespace facebook::react
+```
+
+## Custom structs
+
+You can use the same approach for you custom types in JavaScript such as this one:
+
+```js
+export type CustomType = {
+  key: string,
+  enabled: boolean,
+  time?: number
+};
+```
+
+which can be exposed to your C++ Turbo Native Module via
+
+<Tabs groupId="turbomodule-specs" defaultValue={constants.defaultJavaScriptSpecLanguages} values={constants.javaScriptSpecLanguages}>
+<TabItem value="typescript">
+
+```typescript title="NativeSampleModule.ts"
+   // ...
+   readonly passCustomType: (input: CustomType) => CustomType;
+   // ..
+```
+
+</TabItem>
+<TabItem value="flow">
+
+```js title="NativeSampleModule.js"
+   // ...
+   +passCustomType: (input: CustomType) => CustomType;
+   // ..
+```
+
+</TabItem>
+</Tabs>
+
+### Manually typed
+
+To use this custom type in C++, you need to define your custom Struct and `bridging` function e.g., directly in `NativeSampleModule.h`:
+
+```cpp
+struct CustomType {
+  std::string key;
+  bool enabled;
+  std::optional<int32_t> time;
+
+  bool operator==(const CustomType &other) const {
+    return key == other.key && enabled == other.enabled && time == other.time;
+  }
```

Comment:
We don't.
But users will get a compilation error in case they put those structs into a std::set or set::map.

Let's remove it.
That would be C++ issue and not a TM issue ;-)